src/m365/spo/commands/site/site-admin-list.ts

```diff
@@ -0,0 +1,262 @@
+
+      const requestOptions: CliRequestOptions = {
+        url: `${spoAdminUrl}/_vti_bin/client.svc/ProcessQuery`,
+        headers: {
+          'X-RequestDigest': res.FormDigestValue
+        },
+        data: requestBody
+      };
+
+      const response: string = await request.post<string>(requestOptions);
+      const jsonData: ClientSvcResponse = JSON.parse(response);
+      const responseContent: ClientSvcResponseContents = jsonData[0];
+
+      if (responseContent.ErrorInfo) {
+        throw responseContent.ErrorInfo.ErrorMessage;
+      }
+
+      const sites: SPOSitePropertiesEnumerable = jsonData[jsonData.length - 1];
+      const siteId = sites?._Child_Items_?.[0]?.SiteId ?? undefined;
+
+      if (!siteId) {
+        throw 'Failed to obtain site Id from the provided site URL.';
+      }
+
+      const guid = siteId.replace('/Guid(', '').replace(')/', '');
+
+      if (!validation.isValidGuid(guid)) {
+        throw 'Failed to obtain site Id from the provided site URL.';
+      }
+      return guid;
+    }
+    catch (error) {
+      throw error;
+    }
+  }
+
+  private async getPrimaryAdminLoginNameFromAdmin(adminUrl: string, siteId: string): Promise<string> {
+    try {
+      const requestOptions: CliRequestOptions = {
+        url: `${adminUrl}/_api/SPO.Tenant/sites(%27${siteId}%27)?$select=OwnerLoginName`,
+        headers: {
+          accept: 'application/json;odata=nometadata',
+          'content-type': 'application/json;charset=utf-8'
+        }
+      };
+
+      const response: string = await request.get<string>(requestOptions);
+      const responseContent = JSON.parse(response);
+      return responseContent.OwnerLoginName;
+    }
+    catch (err: any) {
+      throw err;
+    }
+  }
+
+  private async callAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      await logger.logToStderr('Retrieving site administrators...');
+      const requestOptions: CliRequestOptions = {
+        url: `${args.options.siteUrl}/_api/web/siteusers?$filter=IsSiteAdmin%20eq%20true`,
```

Comment:
we should be able  to just leave ` ` instead of %20. It should be encoded for us and is more readable

```suggestion
        url: `${args.options.siteUrl}/_api/web/siteusers?$filter=IsSiteAdmin eq true`,
```

Reply:
Done